### DISCUSSION TITLE
We have a problem with Internationalization (I18n)

### DISCUSSION THREAD

**Discussion by hallgurd** [11ty/eleventy#3211] [Uncategorized]
I want to localize my website to Central Kurdish (CKB) language, but I see that `locale_links` not working, I think it is because this language is not included. Anyone can help me?

**Comment by btrem** [comment]
Can you provide some more info? What program code are you using, what result do you expect, and what result are you getting?

**Reply by hallgurd** [reply]
Structure my website is:
```
├─ src
   └─ en
       ├─ about.njk
       └─ index.njk
   └─ ckb
       ├─ about.njk
       └─ index.njk
```

I want to run code below for creating language switcher: (I use Nunjucks as a template language)
```
{% for link in page.url | locale_links %}
<a href="{{link.url}}" lang="{{link.lang}}" >{{link.label}}</a>
{% endfor %}
```
It should actually return this if current URL is **http://localhost:8080/en/about.html**:
```
<a href="/ckb/about/" lang="ckb" >کوردی</a>
```
But it doesn't return anything **(I only have this problem with this language)**.

In my searches for a solution to this problem, I saw that **I18nPlugin** only uses the languages that come in the **iso-639-1** package, you can see in (https://github.com/11ty/eleventy/blob/main/src/Plugins/I18nPlugin.js).

However, the **Central Kurdish**(ckb) language has been included in the **ISO-639-3**(https://en.wikipedia.org/wiki/ISO_639:c).

I think the **Internationalization (I18n)** plugin should use all **Set** of **ISO 639** code, not just **Set 1**.
See all **Set** here (https://en.wikipedia.org/wiki/ISO_639).

**Reply by btrem** [reply]
I'm way out of my element here, trying to figure things out. Is there some reason that you can't use the iso 639-1 2-letter code `ku`?

**Reply by btrem** [reply]
Replying to myself, it appears to be the difference between northern and central Kurdish. Still, it would help to get some insight from a(n apparently) Sorani speaker.

**Reply by hallgurd** [reply]
Yes, you are right because there is a difference between **Northern** and **Central** Kurdish, I can't use `ku` instead of `ckb`.
**note:** I'm a native speaker of Central Kurdish.

**Reply by jeromecoupe** [reply]
You could roll with a homegrown solution instead of using a plugin. I have used this approach for a few builds and I find it quite flexible. Hope it helps.

https://www.webstoemp.com/blog/multilingual-sites-eleventy/
